Refactoring StatisticalAnalyser class

rmilfont opened the pull request:
Description:Change methods to return parameters separated instead of an array of parameters
Commit:Ref

rmilfont committed:
Refactoring StatisticalAnalyser class
Description:Change methods to return parameters separated instead an array of parameters
Commit:Ref

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
+        return regSizes;
     }
 
     public double linkWeightMean(ArrayList<ArrayList<Path>> paths) {
```
Be more consistent about using mean or average. Both in the method and variable names.

rmilfont replied:
I didn't modify linkWeightMean and pathWeightMean methods that receives an A<A<Paths>>.

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -82,57 +77,63 @@ public double pathWeightMean(ArrayList<ArrayList<Path>> paths) {
-        double acc = 0;
+    public double pathWeightStdDeviation(ArrayList<ArrayList<Path>> paths) {
+        int nPaths = 0;
+        double acc = 0.0; double dev;
```
Avoid putting more than one statement per line.

rmilfont replied:
Done.

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
+    }
 
-        double temp = 0.0;
+    public double linkWeightAverage() {
```
Don't you think averageLinkWeight sound better?

rmilfont replied:
Done for this and some others.

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
+    public double linkWeightStdDeviation() {
+        double temp = 0.0;
+        double mean = linkWeightAverage();
 
```
Get ride of this blank line.

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
 
-        double[] stats = new double[3];
-        double average;
+    private List<Integer> regSizes() {
```
You could find a better name to this method. Something like numberOfRegionsPerVertex or even regionsPerVertex...

rmilfont replied:
Agree.

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -82,57 +77,63 @@ public double pathWeightMean(ArrayList<ArrayList<Path>> paths) {
+        for(ArrayList<Path> alp : paths) {
+            nPaths += alp.size();
+            for(Path path: alp) {
+                dev = path.getWeight() - average;
```
dev could have a more local scope.

rmilfont replied:
Make sense, in fact, in some cases, the remotion of "dev" is better.

jmferreiratech commented on src/StatisticalAnalyser.java:
```diff
@@ -15,55 +15,50 @@ public StatisticalAnalyser(Graph graph, Map<Vertex, ArrayList<Region>> regionsFo
-        double std = 0.0;
+
+    public double linkWeightStdDeviation() {
+        double temp = 0.0;
```
Try to use more meaningful variable names than 'temp', 'sum' and 'acc'.

rmilfont replied:
Done

jmferreiratech reviewed (changes requested):
Please, could you also squash the imports with *? 
You can make a new refactoring commit with the changes requested...

rmilfont committed:
Continue the refactoring at StatisticalAnalyser Class
Description:Change method´s names and parameters for best understanding
Commit:Ref
